### PR TITLE
Replace all icons by styled-icons

### LINKED ISSUE #928: Replace SVG icons by their font-awesome versions
As discussed in https://github.com/opencollective/opencollective-frontend/pull/921#discussion_r225086183 we should migrate `svg` icons from `src/static/icons` to their `font-awesome` equivalents to avoid maintaining svg files and for unification. Most changes will have to be done in JS + CSS.

- [x] `src/static/icons/edit.svg`
- [x] `src/static/icons/external-link.svg`
- [x] `src/static/icons/icon-help-dark.svg`
- [x] `src/static/icons/icon-help-light.svg`
- [x] `src/static/icons/success.svg`
- [x] `src/static/icons/twitter-handler.svg`

### PR DESCRIPTION
**Pull request by Betree**
:information_source: *#921 Should be merged first to apply the same fix on Github icon.*

Resolves #928 by using `styled-icons`.

I tried to keep existing styles as much as possible, there are however some slight changes listed below.

### Edit collective

Modified icons now inherit from block `color` css property, which means their default is now to get colored on links hover.

Before > After

Before After

### HelpTooltip

*Slight change on the `?` icon*

Before > After

Before
After

# Collective cover links

*Twitter icon is now filled*

Before > After

Before
After

# Redeem success

*Thicker check mark*

Before > After

Before
After

### CODE REVIEW

**Comment by Betree** [comment]
Cool ! I need to migrate the new Github icon before merging to master. I'm still in the train with no power plug available :( Will finish this asap

**Comment by Betree** [comment]
@znarf commit rebased and amended with Github icon

Github icon on collective cover

**Comment by cuiki** [comment]
If you have a chance @Betree , you can replace the external link icon with the version that is not outlined: https://fontawesome.com/icons/external-link-square-alt?style=solid

This way it'll be consistent along with the Twitter and GitHub icons.

**Comment by Betree** [comment]
Hi @cuiki the `external-link-square-alt` is not available in the font-awesome version we're currently using (4.7.0). The closest one is [this one](https://fontawesome.com/v4.7.0/icon/external-link-square) which is not exactly the same. 

Please tell me what you think of it and if it's ok for you I'll push the modification:

Peview

**Comment by znarf** [comment]
@Betree we can consider upgrading if necessary.

**Comment by Betree** [comment]
@znarf Upgrading to font-awesome 5 requires some work but seems to bring interesting improvements. It does not use a font anymore, you instead import each svg file independently. It means that you don't bundle unused icons and thus reduce bundle size.

More info here: https://fontawesome.com/how-to-use/on-the-web/using-with/react

I can make the change in this PR if you feel like it is an interesting change or this can go to another issue.

**Comment by znarf** [comment]
@Betree I like the idea of upgrading to the latest version. If we do it, I would say a) either we integrate that in this PR b) either we skip this PR and do it in a new one. 

The only blocker would be for us (or one of the packages we use) to be dependent on the old version and to still have to include it. But I don't think that's the case.

**Comment by Betree** [comment]
I like a) because this PR is already about refactoring the icon system.

**Comment by Betree** [comment]
@znarf I just found the [styled-icons library](https://github.com/jacobwgillespie/styled-icons) which wraps Feather / Font Awesome 5 / Material / Octicons icon packs into Styled Components. It has all the advantages of font-awesome 5 but has more icons bundled and support styled components directly.

They also provide a tool to quickly searching icons available in the library : https://styled-icons.js.org

May I use that instead of the recommended [fort-awesome](https://github.com/FortAwesome/react-fontawesome)?

**Comment by Betree** [comment]
@znarf Replaced all icons (including previously ignored icons in `component/icons`) by their `styled-components` versions. Also removed `font-awesome` dependency as styled-icons brings us `font-awesome 5`.

@cuiki I've been able to use the `external-link-square-alt icon`: Preview


However the existing menu icon in navbar was not available, so I've chosen another one:

* **Existing icon**
Existing icon


* **New icon**

New icon

* **Other icons available**

Material menu
Feather menu

@cuiki Does it look ok to you?
You can check all the icons available here: https://styled-icons.js.org/

**Comment by znarf** [comment]
@Betree this is not building.

File: src/components/CollectiveCover.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ import {
   FormattedTime,
 } from 'react-intl';
 import { get } from 'lodash';
+import { Github } from 'styled-icons/fa-brands/Github.cjs';
```
[the remark this answers is not in the document]
Comment (Betree) [reply src/components/CollectiveCover.js]: I forgot to add the explanations about that, sorry! See this issue: https://github.com/jacobwgillespie/styled-icons/issues/272

**Comment by znarf** [comment]
I don't see the problem with webpack tree shaking, looks good to me like that? What's the report when running `WEBPACK_BUNDLE_ANALYZER=1 npm run build` ?

File: src/components/CollectiveCover.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (HipsterBrown) [reply src/components/CollectiveCover.js]: We're currently using Next.js 7, which includes support for Webpack 4. Maybe we're missing some type of upgrade.

**Comment by Betree** [comment]
@znarf CI passed, up-to-date with master

File: src/components/CollectiveCover.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (znarf) [reply src/components/CollectiveCover.js]: @HipsterBrown I played a bit with webpack recently and might have the answer. What I don't see is how tree shaking affect this issue. Can you tell more precisely what you have in mind?

**Comment by znarf** [comment]
@Betree All right and agree, let's keep it like this and create an issue to document what we ideally want. Issue can be created in that repository.

**Comment by HipsterBrown** [comment]
I initially saw this line about importing CommonJS modules resulting in larger bundles but missed the part about individual imports. 

> Be aware though that importing from the CommonJS icon pack bundles will likely result in significantly larger bundle sizes, because unused icons will be included in the final bundle. If you are unable to configure your bundler to process the ES module bundles, you should import icons individually to avoid large bundles

**Comment by HipsterBrown** [comment]
This looks good to me. Just needs the git conflicts to be resolved. :+1:

**Pull request merged**